### PR TITLE
Add shield iks4a1

### PR DESCRIPTION
**Pull request by avisconti**
x-nucleo-iks4a1 shield is an arduino compatible companion board which can be used on top of Nucleo standard boards for industrial applications. Following MEMS sensor are currently supported:
    
        - LSM6DSO16IS: MEMS 3D accelerometer + 3D gyroscope
        - LSM6DSV16X: MEMS 3D accelerometer + 3D gyroscope
        - LIS2MDL: MEMS 3D magnetometer
        - LPS22DF: Low-power and high-precision MEMS pressure sensor
    
(https://www.st.com/resource/en/data_brief/x-nucleo-iks4a1.pdf)

### CODE REVIEW

File: boards/shields/x_nucleo_iks4a1/doc/index.rst [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+======================================
+
+In this configuration, it is possible to use the equipped Qvar swipe electrode
+(by plugging it on JP6 and JP7 connectors) through the LSM6DSO16IS.
+
+The board configuration is:
+
+ - J4: 3-4 (HUB1_SDx = QVAR1)
+ - J5: 3-4 (HUB1_Scx = QVAR2)
+
```
Comment (erwango) [inline boards/shields/x_nucleo_iks4a1/doc/index.rst:135]: Some words about the specifics of the 3 overlays could be useful and how they interact with each other, if they do.

**Review by erwango — CHANGES_REQUESTED** [review]
Otherwise LGTM

File: boards/shields/x_nucleo_iks4a1/doc/index.rst (hunk 1 above)
Comment (avisconti) [reply boards/shields/x_nucleo_iks4a1/doc/index.rst:135]: OK, I'll try to describe more the three modes (and overlays) that this shield zephyr implementation is providing.

File: samples/shields/x_nucleo_iks4a1/sensorhub1/src/main.c [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+#include <zephyr/device.h>
+#include <zephyr/drivers/sensor.h>
+#include <stdio.h>
+#include <zephyr/sys/util.h>
+
+#ifdef CONFIG_LSM6DSV16X_TRIGGER
+static int lsm6dsv16x_acc_trig_cnt;
+
+static void lsm6dsv16x_acc_trig_handler(const struct device *dev,
+				     const struct sensor_trigger *trig)
```
Comment (fabiobaltieri) [inline samples/shields/x_nucleo_iks4a1/sensorhub1/src/main.c:17]: nit: check the indentation

File: samples/shields/x_nucleo_iks4a1/sensorhub1/src/main.c [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+			    SENSOR_ATTR_SAMPLING_FREQUENCY, &odr_attr) < 0) {
+		printk("Cannot set sampling frequency for LSM6DSV16X accel\n");
+		return;
+	}
+
+	sensor_g_to_ms2(16, &fs_attr);
+
+	if (sensor_attr_set(lsm6dsv16x, SENSOR_CHAN_ACCEL_XYZ,
+			    SENSOR_ATTR_FULL_SCALE, &fs_attr) < 0) {
+		printk("Cannot set fs for LSM6DSV16X accel\n");
```
Comment (fabiobaltieri) [inline samples/shields/x_nucleo_iks4a1/sensorhub1/src/main.c:42]:
```suggestion
		printk("Cannot set full scale for LSM6DSV16X accel\n");
```

(you spell it fully for the other one)

File: boards/shields/x_nucleo_iks4a1/doc/index.rst (hunk 1 above)
Comment (avisconti) [reply boards/shields/x_nucleo_iks4a1/doc/index.rst:137]: ok, added some words... (not too many :D )

File: boards/shields/x_nucleo_iks4a1/doc/index.rst [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+Devicetree Overlays
+*******************
+
+There are three predefined DT overlays in the board:
+
+- :zephyr_file:`boards/shields/x_nucleo_iks4a1/x_nucleo_iks4a1.overlay`
+  This overlay describe sensor connections (and requires h/w configuration to be)
```
Comment (erwango) [inline boards/shields/x_nucleo_iks4a1/doc/index.rst:144]:
```suggestion
  This overlay describes sensor connections (and matching h/w configuration to be done)
```

**Review by erwango — APPROVED** [review]

File: samples/shields/x_nucleo_iks4a1/sensorhub1/src/main.c (hunk 3 above)
Comment (avisconti) [reply samples/shields/x_nucleo_iks4a1/sensorhub1/src/main.c]: ok, done everything you highligthed. Thanks for your patience in reviewing it!

**Review by fabiobaltieri — APPROVED** [review]

**Pull request merged**